src/ol/renderer/canvas/TileLayer.js

```diff
@@ -664,7 +665,12 @@ class CanvasTileLayerRenderer extends CanvasLayerRenderer {
           continue;
         }
       }
-      this.renderComplete = false;
+      if (tileState !== TileState.IDLE) {
+        allIdle = false;
+      }
+      if (tileState !== TileState.ERROR) {
+        this.renderComplete = false;
+      }
```

Comment:
When a tile has settled in the `ERROR` state, it also counts towards a completely rendered map, because it won't be re-fetched.